### DISCUSSION TITLE
How to use socket pub/sub

### DISCUSSION THREAD

**Discussion by mugendi** [kartikk221/hyper-express#134] [Q&A] [answered]
Hello all, I'm building an app via HyperExpress and I'm not sure how to use the pub/sub feature for websockets. 

Tried to find an example without much success. Anyone?

**Comment by kartikk221** [comment, marked as answer]
Hello, please see the `Websocket` component documentation for pub/sub with the MQTT syntax. In simple terms, you would have a flow where whenever a Websocket connection is received, you subscribe to a certain topic and then you can dispatch notifications over those topic(s) with the `Server.publish` method.

**Reply by mugendi** [reply]
Thank you very much @kartikk221 

`server.publish` was the missing link! I was able to figure out the rest but that one part.  The documentation [here](https://github.com/kartikk221/hyper-express/blob/master/docs/Websocket.md) indicates that publish is a method of `ws`, which it is. The trouble I had was that determining what socket to run `ws.publish` from as publishing from a websocket instance, even though it works just fine, means the publishing websocket does not receive the event. Eventually I ended up creating a 'publish only' websocket on the server itself (esssentially a server-server websocket) and using that to `ws.publish` to all other client sockets. But that felt very wrong, even though it worked, so that's why I asked.

Turns out there is a `server.publish` method too! Thank you so much. I think this needs to be documented. I can create a PR when I have the time just to document that bit.

**Reply by kartikk221** [reply]
Oh I see, will make a note of that in the documentation for future changes.
